### PR TITLE
ddl: fix inappropriate schema state output by admin show ddl jobs

### LINKED ISSUE #24420: ddl: the schema state is not appropriate for some removed-type ddl
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Screenshot_20210506_142031

As you see in the pic above
state-none has two meanings:
1. one for adding-type DDL, the job hasn't started yet / the job has been rollbacked.
2. one for removed-type DDL, the job hasn't started yet / the job has done.

### 2. What did you expect to see? (Required)
For some of those, queueing has a good semantic explanation, but not for the others.

### 3. What did you see instead (Required)
None/Queueing

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #24420, related to #23494

Problem Summary:

The information "queueing" is introduced by https://github.com/pingcap/parser/pull/1210.

> state-none has two meanings:
>
> - one for adding-type DDL, the job hasn't started yet / the job has been rollbacked.
> - one for removed-type DDL, the job hasn't started yet / the job has done.

### What is changed and how it works?

This PR picks up the work from #31544.

Before this PR:
```
+--------+---------+------------+---------------+----------------------+-----------+----------+-----------+---------------------+---------------------+---------------------+---------+
| JOB_ID | DB_NAME | TABLE_NAME | JOB_TYPE      | SCHEMA_STATE         | SCHEMA_ID | TABLE_ID | ROW_COUNT | CREATE_TIME         | START_TIME          | END_TIME            | STATE   |
+--------+---------+------------+---------------+----------------------+-----------+----------+-----------+---------------------+---------------------+---------------------+---------+
|    166 | sbtest  | sbtest1    | add index     | write reorganization |       160 |      162 |         0 | 2022-04-11 14:47:10 | 2022-04-11 14:47:10 | NULL                | running |
|    167 | sbtest2 | sbtest1    | add index     | queueing             |       158 |      164 |         0 | 2022-04-11 14:47:10 | NULL                | NULL                | none    |
```

After this PR:
```
mysql> admin show ddl jobs;
+--------+---------+------------+---------------+----------------------+-----------+----------+-----------+---------------------+---------------------+---------------------+----------+
| JOB_ID | DB_NAME | TABLE_NAME | JOB_TYPE      | SCHEMA_STATE         | SCHEMA_ID | TABLE_ID | ROW_COUNT | CREATE_TIME         | START_TIME          | END_TIME            | STATE    |
+--------+---------+------------+---------------+----------------------+-----------+----------+-----------+---------------------+---------------------+---------------------+----------+
|    154 | sbtest  | sbtest1    | add index     | write reorganization |       146 |      150 |    230400 | 2022-04-11 14:24:59 | 2022-04-11 14:24:59 | NULL                | running  |
|    155 | sbtest2 | sbtest1    | add index     | none                 |       148 |      152 |         0 | 2022-04-11 14:25:00 | NULL                | NULL                | queueing |
```

Basically, there are 2 changes:
- There is no "queueing" `SCHEMA_STATE` anymore. Even if the job is in the queue, the `SCHEMA_STATE` is "none".
- The state "queueing" has been added to the `STATE` column. It is used to distinguish whether a job is waiting in the DDL queue.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

See the result above.

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix an issue that the schema state output by 'admin show ddl jobs' is inappropriate.
```

### CODE REVIEW

**Review by wjhuang2016 — COMMENTED** [review]
What's the manual test and its result?

File: ddl/ddl_worker.go [hunk 1]
```diff
@@ -623,6 +624,10 @@ func (w *worker) handleDDLJobQueue(d *ddlCtx) error {
 				return errors.Trace(err)
 			}
 
+			if job.IsQueueing() {
+				job.State = model.JobStateNone
```
Comment (zimulala) [inline ddl/ddl_worker.go:628]: Could we remove this logic? We do `job.State = model.JobStateRunning` in `runDDL`.

Comment (tangenta) [reply ddl/ddl_worker.go:628]: If we don't set it here, the state observed by the `OnJobRunBefore` callback would be "queueing". There are a few tests depend on this.

**Comment by tangenta** [comment]
@wjhuang2016 I have update the manual test result. The difference is shown in the last column.

File: ddl/ddl_worker.go (hunk 1 above)
Comment (zimulala) [reply ddl/ddl_worker.go:628]: I don't think it's necessary to change the code for testing. It's already in `OnJobRunBefore`, meaning `runDDL` hasn't been called yet, so I think the `queueing` is understandable.

**Review by zimulala — APPROVED** [review]
LGTM

**Comment by zimulala** [comment]
Please add `Release note` for this PR.

**Comment by easonn7** [comment]
LGTM

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**